lib/cfa/base_model.rb

```diff
@@ -115,14 +128,20 @@ def modify(key, value)
       true
     end
 
+    # Replace a commented out entry and return `true`,
+    # or do nothing and return `false`.
+    # @return [Boolean]
     def uncomment(key, value)
       # Try to find if it is commented out, so we can replace line
       matcher = Matcher.new(
         collection:    "#comment",
+        # FIXME: this assumes a specific "=" syntax, bypassing the lens
+        # FIXME: this will match also "# If you set FOO=bar then..."
         value_matcher: /(\s|^)#{key}\s*=/
       )
       return false unless  data.data.any?(&matcher)
 
+      # FIXME: this assumes that *data* is an AugeasTree
```

Comment:
Yes, it is reasonable, I just wanted to point out what I think is the only place where this is assumed in `BaseModel`. Otherwise its API hides the details of `data` well.